Framework/Geometry/src/Instrument.cpp

```diff
@@ -1235,5 +1237,15 @@ Instrument::ContainsState Instrument::containsRectDetectors() const {
 
 } // containsRectDetectors
 
+/// Only for use by ExperimentInfo. Returns the pointer to the DetectorInfo.
+const Beamline::DetectorInfo *Instrument::detectorInfo() const {
+  return m_detectorInfo;
+}
+
+/// Only for use by ExperimentInfo. Sets the pointer to the DetectorInfo.
+void Instrument::setDetectorInfo(const Beamline::DetectorInfo *detectorInfo) {
+  m_detectorInfo = detectorInfo;
```

Comment:
Not a resource leak, the `DetectorInfo` is owned by an `ExperimentInfo`. It definitely is a temporary change, but it will take a while until the Instrument-2.0 implementation has progressed far enough to remove it.

You are right that it is unsafe: The pointer could in principle get stale if the workspace owning the `DetectorInfo` is deleted. I believe that this is unlikely given the way Mantid deals with things: (1) We never have a stand-alone instrument, apart from short-lived ones within an algorithm/test. (2) Workspaces are rarely explicitly deleted but just go out of scope when an algorithm/test ends. Do you think the risk is too high?